plugin/pkg/policy/engine.go

```diff
@@ -0,0 +1,71 @@
+package policy
+
+import (
+	"context"
+
+	"github.com/coredns/coredns/request"
+)
+
+const (
+	// TypeNone a no value
+	TypeNone = byte(iota)
+	// TypeRefuse policy action is REFUSE (do not resolve a query and return rcode REFUSED)
+	TypeRefuse
+	// TypeAllow policy action is ALLOW (continue to resolve query)
+	TypeAllow
+	// TypeBlock policy action is BLOCK (do not resolve a query and return rcode NXDOMAIN)
+	TypeBlock
+	// TypeDrop policy action is DROP (do not resolve a query and simulate a lost query)
+	TypeDrop
+
+	// TypeCount total number of actions allowed
+	TypeCount
+)
+
+const (
+	nameTypeNone   = "none"
+	nameTypeAllow  = "allow"
+	nameTypeRefuse = "refuse"
+	nameTypeBlock  = "block"
+	nameTypeDrop   = "drop"
+)
+
+// NameTypes keep a mapping of the byte constant to the corresponding name
+var NameTypes [TypeCount]string
+
+func initNameTypes() {
+	NameTypes[TypeNone] = nameTypeNone
+	NameTypes[TypeRefuse] = nameTypeRefuse
+	NameTypes[TypeAllow] = nameTypeAllow
+	NameTypes[TypeBlock] = nameTypeBlock
+	NameTypes[TypeDrop] = nameTypeDrop
+}
+
+func init() {
```

Comment:
this shouldn't be done in an init function

Reply:
and where then ? it is a Map and cannot be created as a constant.